console/console-init/ui/src/modules/iot-device/DeviceListPage.tsx

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import { useDocumentTitle } from "use-patternfly";
+import {
+  Grid,
+  GridItem,
+  Card,
+  CardBody,
+  DropdownItem
+} from "@patternfly/react-core";
+import { ISortBy } from "@patternfly/react-table";
+import {
+  DeviceFilter,
+  DeviceListAlert,
+  DeviceListFooterToolbar,
+  DeviceListToolbar,
+  IDevice,
+  IDeviceFilter
+} from "modules/iot-device/components";
+import { DeviceListContainer } from "modules/iot-device/containers";
+import { compareObject } from "utils";
+import { getInitialFilter } from "modules/iot-device/utils";
+
+export default function DeviceListPage() {
+  useDocumentTitle("Device List");
+
+  const [totalDevices, setTotalDevices] = useState<number>(0);
+  const [isAllSelected, setIsAllSelected] = useState<boolean>(false);
+  const [perPage, setPerPage] = useState<number>(10);
+  const [selectedDevices, setSelectedDevices] = useState<IDevice[]>([]);
+  const [filter, setFilter] = useState<IDeviceFilter>(getInitialFilter());
+  const [appliedFilter, setAppliedFilter] = useState<IDeviceFilter>(
+    getInitialFilter()
+  );
+
+  const onSelectDevice = (
+    data: IDevice,
+    isSelected: boolean,
+    isAllSelected?: boolean
+  ) => {
+    if (isSelected === true && selectedDevices.indexOf(data) === -1) {
+      setSelectedDevices(prevState => [...prevState, data]);
+    } else if (isSelected === false) {
+      setSelectedDevices(prevState =>
+        prevState.filter(
+          device =>
+            !compareObject(
+              {
+                id: device.deviceId
+              },
+              {
+                id: data.deviceId
+              }
+            )
+        )
+      );
+    }
+    if (isAllSelected) {
+      setIsAllSelected(true);
+    }
+  };
+
+  const [sortDropDownValue, setSortDropdownValue] = useState<ISortBy>();
+
+  const onSelectAll = (val: boolean) => {
+    onSelectAllDevices(val);
+  };
+
+  const onSelectAllDevices = (isSelected: boolean) => {
+    if (isSelected === true) {
+      setIsAllSelected(true);
+    } else if (isSelected === false) {
+      setIsAllSelected(false);
+      setSelectedDevices([]);
+    }
+  };
+
+  const runFilter = () => {
+    setAppliedFilter(filter);
+  };
+
+  const selectAllDevices = (devices: IDevice[]) => {
+    setSelectedDevices(devices);
+  };
+
+  const onSelectEnableDevices = () => {
+    // TO BE DONE AFTER BACKEND IS READY
```

Comment:
For Todo task please add comment as  //TODO: To be done after backend is ready. So that we can do quick verification of TODO